packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -505,29 +467,72 @@ export class WalletFactory {
         return false;
     }
 
-    private isShadowSignerEnabled(chain: Chain, options?: WalletOptions): boolean {
-        return (chain === "solana" || chain === "stellar") && options?.shadowSigner?.enabled !== false;
+    private isShadowSignerEnabled<C extends Chain>(
+        chain: C,
+        adminSigner: SignerConfigForChain<C>,
+        delegatedSigners: Array<SignerConfigForChain<C>> = []
+    ): boolean {
+        const ncSigners = [adminSigner, ...delegatedSigners].filter(
+            (signer) => signer.type === "email" || signer.type === "phone"
+        ) as Array<EmailSignerConfig | PhoneSignerConfig>;
+        return (
+            !this.apiClient.isServerSide &&
+            (chain === "solana" || chain === "stellar") &&
+            ncSigners.length > 0 &&
+            ncSigners.some((signer) => signer.shadowSigner?.enabled !== false)
+        );
+    }
+
+    private async buildDelegatedSigners<C extends Chain>(
+        adminSigner: SignerConfigForChain<C>,
+        args: WalletCreateArgs<C>
+    ): Promise<{
+        delegatedSigners: Array<DelegatedSigner | RegisterSignerParams | { signer: PasskeySignerConfig }>;
+        shadowSignerPublicKey: string | null;
+        shadowSignerPrivateKey: CryptoKey | null;
+    }> {
+        const {
+            delegatedSigners: updatedDelegatedSigners,
```

Comment:
i would keep this called as delegatedSigners and the one below can have a more specific name for the changes/enhancements we do to signers